ls1intum/Artemis
# `Development`: Remove client compilation from server and test run configurations, and add default analytics setting

### Checklist
Not applicable

### Motivation and Context


In #4022 the build configuration changed quite a lot resulting in the server application compiling the client on startup. This takes a lot of unnecessary time especially since one starts the client separately anyway.
Additionally, Angular asks on the first start whether one wants to send analytics or not. Depending on what git client you use, it can get annoying having such changes around.

### Description

- I updated the server run configurations to not use the default build step but use the gradle goal. I then had to remove the webapp compilation and the testing steps
- I also removed the webapp compilation from the default gradle task and therefore added a new configuration file
- In the process, I whitelisted the run configurations folder in the .gitignore. Keep in mind that this doesn't mean that everyone's run configurations get shared. If you want to share a run configuration over VCS you have to manually tick a box
- Using this approach starts bringing in the modernized plugin which was part of our configuration but never used. We decided to keep it but it has low priority. I, therefore, made sure that the warnings won't make the system exit. This has to be dealt with in #4312 
- I set the default value for sending analytics to false in the Angular configuration. The reason why I decided to set it in the first place is described above. The reason why I set it to false is that sending data to a third party should never be the default.

### Steps for Testing

**If you think we could remove more steps from application start, let me now.**

Prerequisites:
Your IDE with Artemis loaded

Server run configurations
1. Checkout the branch of this PR if not already done
2. Use the run configuration dropdown to select one server application configuration (not Server & Client) [1]
3. Start it by clicking on the green play button
4. Make sure that no npm, node, webapp, etc task gets executed
5. Make sure the application gets fully started
6. Restart the application, it should start now within a few seconds due to caching. Depending on how much was cached before, it might have been already very fast the first time

Tests
1. Select any server test with a right-click and execute it. It can be one test case, a file, a class, a package or even the whole test suite. It has to be executed from the IDE though, not the gradle tasks or the CLI since that's what we try to test
2. Make sure that no npm, node, webapp, etc task gets executed
3. Make sure the test(s) get(s) fully executed
4. Restart the test(s), it/they should start now within 20 seconds due to caching. Depending on how much was cached before, it might have been already very fast the first time

Client run configuration
1. Start the Client run configuration from the dropdown [1]
2. After starting `ng serve --hmr`, no Angular blocker should appear asking you for your consent [2]

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots


[1]
![](https://gyazo.com/2c450e8adc33d084fd6790cdee99a4db/raw)
[2]
![](https://gyazo.com/978abf78f90ef3891f63b471a0486395/raw)
